internal/tracing/attributes.go

```diff
@@ -56,31 +60,37 @@ func (s SpanOptions) EventOptions() []trace.EventOption {
 // For convenience, it's returned as a SpanOptions object to allow it to be
 // passed directly to the wrapping helper methods in this package such as
 // SpanWrapFunc.
-func ProjectOptions(proj *types.Project) SpanOptions {
+func ProjectOptions(ctx context.Context, proj *types.Project) SpanOptions {
 	if proj == nil {
 		return nil
 	}
-
 	capabilities, gpu, tpu := proj.ServicesWithCapabilities()
 	attrs := []attribute.KeyValue{
 		attribute.String("project.name", proj.Name),
 		attribute.String("project.dir", proj.WorkingDir),
 		attribute.StringSlice("project.compose_files", proj.ComposeFiles),
-		attribute.StringSlice("project.services.active", proj.ServiceNames()),
-		attribute.StringSlice("project.services.disabled", proj.DisabledServiceNames()),
 		attribute.StringSlice("project.profiles", proj.Profiles),
 		attribute.StringSlice("project.volumes", proj.VolumeNames()),
 		attribute.StringSlice("project.networks", proj.NetworkNames()),
 		attribute.StringSlice("project.secrets", proj.SecretNames()),
 		attribute.StringSlice("project.configs", proj.ConfigNames()),
 		attribute.StringSlice("project.extensions", keys(proj.Extensions)),
 		attribute.StringSlice("project.includes", flattenIncludeReferences(proj.IncludeReferences)),
+		attribute.StringSlice("project.services.active", proj.ServiceNames()),
+		attribute.StringSlice("project.services.disabled", proj.DisabledServiceNames()),
 		attribute.StringSlice("project.services.build", proj.ServicesWithBuild()),
 		attribute.StringSlice("project.services.depends_on", proj.ServicesWithDependsOn()),
 		attribute.StringSlice("project.services.capabilities", capabilities),
 		attribute.StringSlice("project.services.capabilities.gpu", gpu),
 		attribute.StringSlice("project.services.capabilities.tpu", tpu),
 	}
+	countExtends := 0
+	metrics, ok := ctx.Value(Metrics{}).(Metrics)
+	if ok {
+		countExtends = metrics.CountExtends
+	}
+	attrs = append(attrs, attribute.Int("project.services.extends", countExtends))
```

Comment:
shall we skip this attribute if we don't have a value? Using `0` might corrupt the metric

Reply:
We could only append if the `Metrics{}` is present inside the context, but I'm sure the 0 default value we'll be used on the other side